### PR TITLE
This tail refactor turned into an organ refactor. Funny how that works.

### PR DESCRIPTION
**Pull request by Kapu1178**
### So yeah, uh, we have /organ/internal now.
External organs are cosmetic, don't process, and are stored on limbs directly. Most cannot be surgically removed.
Internal organs process, and are stored in nullspace until the limb they technically belong to is dismembered. ~~This is legacy behavior, fuck you, I'll change it in a future PR.~~ This is intentional behavior, apparently.

- Rewrites tails to be external_organs instead of whatever the fuck they are now. Changes tail wagging to a simple signal.
- Rewrites how external_organs handle color, now defined by the organ itself. Organs can choose to use no color, their own color, hair color, or the parent limb color.
- Stores a list of all external_organs on /carbon, mostly to prevent nested for loop hell.
- External organ color now behaves like Kapulimbs, meaning they will only change color when forced to.
- External organs are now handled (almost) identically to standard organs, and can be surgically removed if flagged.
- External organs are now responsible for their appearance on Insert()
- External organs no longer sit in nullspace forever if their parent is deleted. I don't know how this issue didn't come up sooner.

## Why It's Good For The Game
Brings more ancient shitcode inline with modern, less shit code.

### CODE REVIEW

**Review by Time-Green — APPROVED** [review]
Yes… yes… very good

**Comment by Profakos** [comment]
Could you please add ORGAN_COLOR_FACEHAIR flag for future proofing? Thank you... (a week ago on our downstream we changed snouts to take facial hair as a colour, we have quite a bit of a conflict resolution on our hands in the near future, haha)

Paging @VioletN  so she'll see this.

**Comment by MTandi** [comment]
You can still spin people with tail around as a Hulk, right?

**Review by Maurukas — APPROVED** [review]
Mapping is a simple repath and is fine.

I believe that `dna.features` is evil and only misfortune can come from looking at it. So one of the code maintainers can do that.

File: code/datums/dna.dm [hunk 1]
```diff
@@ -195,8 +195,10 @@ GLOBAL_LIST_INIT(total_uf_len_by_block, populate_total_uf_len_by_block())
 		L[DNA_ETHEREAL_COLOR_BLOCK] = sanitize_hexcolor(features["ethcolor"], include_crunch = FALSE)
 	if(features["body_markings"])
 		L[DNA_LIZARD_MARKINGS_BLOCK] = construct_block(GLOB.body_markings_list.Find(features["body_markings"]), GLOB.body_markings_list.len)
+	if(features["tail_cat"])
+		L[DNA_TAIL_BLOCK] = construct_block(GLOB.tails_list_human.Find(features["tail_cat"]), GLOB.tails_list_human.len)
 	if(features["tail_lizard"])
-		L[DNA_LIZARD_TAIL_BLOCK] = construct_block(GLOB.tails_list_lizard.Find(features["tail_lizard"]), GLOB.tails_list_lizard.len)
+		L[DNA_LIZARD_TAIL_BLOCK] = construct_block(GLOB.tails_list_lizard.Find(features["tail_lizard"]), GLOB.tails_list_human.len)
```
Comment (Kapu1178) [inline code/datums/dna.dm:201]:
```suggestion
		L[DNA_LIZARD_TAIL_BLOCK] = construct_block(GLOB.tails_list_lizard.Find(features["tail_lizard"]), GLOB.tails_list_lizard.len)
```

**Review by Ghilker — COMMENTED** [review]
mostly docs

File: code/modules/surgery/organs/_organ.dm [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+	var/high_threshold_cleared
+	var/low_threshold_cleared
+
+	///When you take a bite you cant jam it in for surgery anymore.
+	var/useable = TRUE
+	var/list/food_reagents = list(/datum/reagent/consumable/nutriment = 5)
+	///The size of the reagent container
+	var/reagent_vol = 10
+
+	var/failure_time = 0
```
[the remark this answers is not in the document]
Comment (Kapu1178) [reply code/modules/surgery/organs/_organ.dm:38]: I didn't make these. The file was simply moved. I do not have any intention to code dive organs to doc vars I didn't create.

File: code/modules/surgery/organs/_organ.dm [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+			return high_threshold_passed
+		if(damage > low_threshold && prev_damage <= low_threshold)
+			return low_threshold_passed
+	else
+		if(prev_damage > low_threshold && damage <= low_threshold)
+			return low_threshold_cleared
+		if(prev_damage > high_threshold && damage <= high_threshold)
+			return high_threshold_cleared
+		if(prev_damage == maxHealth)
+			return now_fixed
```
[the remark this answers is not in the document]
Comment (Kapu1178) [reply code/modules/surgery/organs/_organ.dm:198]: I didn't make this and would rather not fuck it up.

File: code/modules/surgery/organs/external/_external_organs.dm [hunk 4]
```diff
@@ -48,78 +55,130 @@
...
 	if(ownerlimb)
-		ownerlimb.external_organs.Remove(src)
-		ownerlimb.contents.Remove(src)
-		if(external_bodytypes) //Happens after removal from contents, and before ownerlimb is null.
-			ownerlimb.synchronize_bodytypes(organ_owner)
-		ownerlimb = null
+		remove_from_limb()
 
-	organ_owner.update_body_parts()
+	if(organ_owner)
```
[the remark this answers is not in the document]
Comment (Kapu1178) [reply code/modules/surgery/organs/external/_external_organs.dm:98]: Doesn't make it any more readable.

File: code/modules/surgery/organs/_organ.dm [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+INITIALIZE_IMMEDIATE(/obj/item/organ)
+
+/obj/item/organ/Initialize(mapload)
+	. = ..()
+	if(organ_flags & ORGAN_EDIBLE)
+		AddComponent(/datum/component/edible,\
+			initial_reagents = food_reagents,\
+			foodtypes = RAW | MEAT | GROSS,\
+			volume = reagent_vol,\
+			after_eat = CALLBACK(src, .proc/OnEatFrom))
```
[the remark this answers is not in the document]
Comment (Kapu1178) [reply code/modules/surgery/organs/_organ.dm:55]: Didn't write it, not my problem

File: code/modules/surgery/bodyparts/dismemberment.dm [hunk 6]
```diff
@@ -306,14 +322,15 @@
 	if(!.) //If it failed to replace, re-attach their old limb as if nothing happened.
 		old_limb.attach_limb(limb_owner, TRUE)
 
-/obj/item/bodypart/head/replace_limb(mob/living/carbon/head_owner, special)
+/*/obj/item/bodypart/head/replace_limb(mob/living/carbon/head_owner, special)
```
[the remark this answers is not in the document]
Comment (Kapu1178) [reply code/modules/surgery/bodyparts/dismemberment.dm:325]: Oops

File: code/modules/surgery/organs/_organ.dm (hunk 5 above)
[the remark this answers is not in the document]
Comment (Ghilker) [reply code/modules/surgery/organs/_organ.dm:55]: You moved it tho, is just an early return even

File: code/__DEFINES/dcs/signals/signals_mob/signals_mob_living.dm [hunk 7]
```diff
@@ -6,9 +6,11 @@
 // Organ signals
 /// Called on the organ when it is implanted into someone (mob/living/carbon/receiver)
 #define COMSIG_ORGAN_IMPLANTED "comsig_organ_implanted"
-
+/// Called when using the *wag verb
```
[the remark this answers is not in the document]
Comment (Kapu1178) [reply code/__DEFINES/dcs/signals/signals_mob/signals_mob_living.dm:9]: It's an emote. They're datums.

[the remark this answers is not in the document]
Comment (Ghilker) [reply code/__DEFINES/dcs/signals/signals_mob/signals_mob_living.dm:9]: Then add /datum/emote/wag or whatever it is

**Comment by Ghilker** [comment]
I still think you should doc whatever you touch, since is in the guidelines, but maints will decide on this and if they don't care I don't care

**Review by Kylerace — APPROVED** [review]
after 40 minutes im going insane. looks fine ill look again tomorrow:tm:

**Comment by Kapu1178** [comment]
I assumed this would be testmerged atleast once I offload all blame onto Kyler if something breaks.

**Pull request merged**